Firestore/core/src/firebase/firestore/auth/firebase_credentials_provider_apple.mm

```diff
@@ -120,19 +120,26 @@
     }
   };
 
+  bool force_refresh = false;
+  {
+    std::lock_guard<std::mutex> lock{contents_->mutex};
+    force_refresh = contents_->force_refresh;
+    contents_->force_refresh = false;
+  }
+
   // TODO(wilhuff): Need a better abstraction over a missing auth provider.
   if (contents_->auth) {
-    [contents_->auth getTokenForcingRefresh:contents_->force_refresh
+    [contents_->auth getTokenForcingRefresh:force_refresh
                                withCallback:get_token_callback];
   } else {
     // If there's no Auth provider, call back immediately with a nil
     // (unauthenticated) token.
     get_token_callback(nil, nil);
   }
-  contents_->force_refresh = false;
 }
 
 void FirebaseCredentialsProvider::InvalidateToken() {
+  std::lock_guard<std::mutex> lock{contents_->mutex};
```

Comment:
I presume this is also necessary.

Reply:
On the surface it seems this shouldn't be necessary since `InvalidateToken` and `GetToken` should only be called from the async queue.

However, I can also see how this might be considered a race by tsan because we're accessing this object with a lock held in some cases but not in others.

Meanwhile it's definitely easier to reason about this code if the contents of `contents_` shouldn't be accessed without holding its lock.

However, that line of reasoning suggests that unprotected access to `contents_->token_counter` (and possibly `contents_->auth`) should also be fixed.

Reply:
Since this wasn't introduced by gRPC changes, I think I'll just keep this change off this PR for now and upload it separately later on.